bindings/wasm/manifold.d.ts

```diff
@@ -12,474 +12,32 @@
- * manifold. If the polygons cross the Y-axis, only the part on the positive X
- * side is used. Geometrically valid input will result in geometrically valid
- * output.
- *
- * @param crossSection A set of non-overlapping polygons to revolve.
- * @param circularSegments Number of segments along its diameter. Default is
- * calculated by the static Defaults.
- */
-function revolve(crossSection: Polygons, circularSegments?: number): Manifold;
-
-function union(a: Manifold, b: Manifold): Manifold;
-function difference(a: Manifold, b: Manifold): Manifold;
-function intersection(a: Manifold, b: Manifold): Manifold;
-
-function union(manifolds: Manifold[]): Manifold;
-function difference(manifolds: Manifold[]): Manifold;
-function intersection(manifolds: Manifold[]): Manifold;
-
-/**
- * Constructs a new manifold from a list of other manifolds. This is a purely
- * topological operation, so care should be taken to avoid creating
- * overlapping results. It is the inverse operation of Decompose().
- *
- * @param manifolds A list of Manifolds to lazy-union together.
- */
-function compose(manifolds: Manifold[]): Manifold;
-
-/**
- * Constructs a level-set Mesh from the input Signed-Distance Function (SDF).
- * This uses a form of Marching Tetrahedra (akin to Marching Cubes, but better
- * for manifoldness). Instead of using a cubic grid, it uses a body-centered
- * cubic grid (two shifted cubic grids). This means if your function's interior
- * exceeds the given bounds, you will see a kind of egg-crate shape closing off
- * the manifold, which is due to the underlying grid.
- *
- * @param sdf The signed-distance function which returns the signed distance of
- * a given point in R^3. Positive values are inside, negative outside.
- * @param bounds An axis-aligned box that defines the extent of the grid.
- * @param edgeLength Approximate maximum edge length of the triangles in the
- * final result. This affects grid spacing, and hence has a strong effect on
- * performance.
- * @param level You can inset your Mesh by using a positive value, or outset
- * it with a negative value.
- */
-function levelSet(
-    sdf: (point: Vec3) => number, bounds: Box, edgeLength: number,
-    level?: number): Manifold;
-
-/**
- * @name Defaults
- * These static properties control how circular shapes are quantized by
- * default on construction. If circularSegments is specified, it takes
- * precedence. If it is zero, then instead the minimum is used of the segments
- * calculated based on edge length and angle, rounded up to the nearest
- * multiple of four. To get numbers not divisible by four, circularSegments
- * must be specified.
- */
-///@{
-function setMinCircularAngle(angle: number): void;
-function setMinCircularEdgeLength(length: number): void;
-function setCircularSegments(segments: number): void;
-function getCircularSegments(radius: number): number;
-///@}
+import * as T from './manifold-encapsulated-types';
+export * from './manifold-global-types';
```

Comment:
Good idea on separating these. I definitely don't want to go back to the `declare` version. The first priority is getting the types right for npm users - the editor is our own problem. Maybe we could hack it a little though - I'm considering some string concatenation and replacing `export` with `declare` during import to the editor.

Reply:
I'm not sure if that would work, because Typescript really hates bundling type definitions, but it's worth a try. Maybe you could use something like dts-bundle?